sqlglot/optimizer/simplify.py

```diff
@@ -430,6 +432,74 @@ def simplify_parens(expression):
     return expression
 
 
+CONSTANTS = (
+    exp.Literal,
+    exp.Boolean,
+    exp.Null,
+)
+
+
+def simplify_coalesce(expression):
+    # COALESCE(x) -> x
+    if (
+        isinstance(expression, exp.Coalesce)
+        and not expression.expressions
+        # COALESCE is also used as a Spark partitioning hint
+        and not isinstance(expression.parent, exp.Hint)
+    ):
+        return expression.this
+
+    if not isinstance(expression, COMPARISONS):
+        return expression
+
+    if isinstance(expression.left, exp.Coalesce):
+        coalesce = expression.left
+        other = expression.right
+    elif isinstance(expression.right, exp.Coalesce):
+        coalesce = expression.right
+        other = expression.left
+    else:
+        return expression
+
+    # COALESCE(x, y, z)
+    #                ^ grab that guy
+    coalesce_args = coalesce.expressions
+    if coalesce_args:
+        rightmost_arg = coalesce_args[-1]
+    else:
+        return expression
```

Comment:
Isn't it guaranteed that `coalesce.expressions` will always contain ≥1 items here?

Reply:
Because of the first line in the function?

I'm cheating and using one function to both simplify COALESCE and comparisons containing a COALESCE. So this is a different expression.

Reply:
I'm not sure I follow, isn't `coalesce` set to the `Coalesce` expression in the `if-elif-else` branch right above?

Reply:
`COALESCE(x) = 1`

Reply:
I guess it depends on whether we transform in post-order or pre-order

Reply:
Aah the piece I missed was that `Coalesce` is one of those expressions where you have both `this` and `expressions`. Makes sense.